Another attempt with UI-forms

klees opened the pull request:
.. as alternative to #1731

klees committed:
SP: switched settings form to ui-framework

nhaagen commented on Modules/StudyProgramme/classes/class.ilObjStudyProgrammeSettingsGUI.php:
```diff
@@ -159,7 +178,7 @@ protected function update() {
 			if($update_possible) {
```
$update_possible = $this->checkForm($form);
or set in the check for (!is_null($content)

nhaagen reviewed (approved):
I like this a lot. Incorporated the changes into LearningSequence already ;)

klees committed:
SP: correction in form-handling

klees committed:
SP: minor wording in comment

klees committed:
SP: use text-area for description in settings

klees committed:
SP: Status is a required setting

shecken committed:
Activate meta data for iass

shecken committed:
Use object metadata gui

shecken committed:
IAss: Fix user autocomplete

shecken committed:
IAss: Presentation Talbe first steps

shecken committed:
Show meta data on info tab

chfsx committed:
removed duplicate files and naming

Amstutz committed:
Merge pull request #1146 from studer-raimann/feature/5-4/main-menu-services-interface

Feature/5 4/main menu services interface

mjansenDatabay committed:
Manual Mail Templates: Moved service to DIC

xus committed:
Centralize News Settings

Missing:
- change the links to the News settings and remove all links to the contextBlockGUI.
- Remove unused methods

klees committed:
Merge pull request #1202 from klees/trunk_kiosk_docu

 KioskMode: enhanced docu (#23574) and fixed param to TOCBuilder

xus committed:
news block - link to context object settings

xus committed:
Merge branch '54_new_keywords' into 54_keywords_materials_settings

xus committed:
Merge branch '54_centralising_news_settings' into 54_keywords_materials_settings

Commit:
Revert "Add subfolders to Forum classes"

This reverts commit 8933056b586a72366c95e25a2fdf18c217112b49.

Commit:
Revert "Remove some Forum requirements"

This reverts commit 4d98a798a9f8813b849a194b20c3d4c352c4e59f.

Commit:
Composer

mjansenDatabay committed:
Merge pull request #1226 from legionth/revers-forum-commits

[Forum] Reverse forum subfolder commits

smeyer-ilias committed:
Added presenter for custom icons

smeyer-ilias committed:
Added presenter for custom icons

smeyer-ilias committed:
Usage for custom icon presenter

smeyer-ilias committed:
Usage for custom icon presenter

mjansenDatabay committed:
Forum: ilCtrl structure reload

mjansenDatabay committed:
Forum: Fixed #23717

Amstutz committed:
UI: Proposal for Outlined Icons

jbraun-ilias committed:
fixed #23684

jbraun-ilias committed:
fix #20790

mjansenDatabay committed:
Mail: Fixed translations for placeholder

mjansenDatabay committed:
Terms of Service: Fixed wrong ID

xus committed:
add multicommand + remove specific items

rsheer committed:
Refactored WebDAV for unit testing

rsheer committed:
Added some unit tests for webdav

xus committed:
fix array of items to update

mjansenDatabay committed:
Merge pull request #1182 from mjansenDatabay/feature-neutral-gender

User Service: Neutral Gender

rsheer committed:
Merge branch 'trunk' of https://github.com/ILIAS-eLearning/ILIAS into trunk

xus committed:
remove the checked if stored in db

rsheer committed:
Removed the WebDAV-in-Browser feature and exchanged it for a redirect to the mount-instruction page

shecken committed:
IAss: New lang vars

xus committed:
First steps with filters

mjansenDatabay committed:
Merge branch 'trunk' into feature-content-page-kiosk-mode

mjansenDatabay committed:
Content Page: Tiny changes regarding class names for ilCtrl

mjansenDatabay committed:
Content Page: Kiosk mode

mjansenDatabay committed:
Mail: Delivered single file instead of *.zip for a single attachment / Changed column width and column title

shecken committed:
IAss: Finish implementation of presentation table

mjansenDatabay committed:
Content Page: Fixed target

shecken committed:
IAss: Add PHP7 typehinting and fixes

xus committed:
applyFilter + resertFilter + filter by title

mjansenDatabay committed:
Manual Mail Templates: Fixed initial selection

mjansenDatabay committed:
Merged latest trunk changes

smeyer-ilias committed:
Added new dependency for ReferencePresenter; removed DB dependency for IconFactory; Added parameter/return type hints

smeyer-ilias committed:
Added new dependency for ReferencePresenter; removed DB dependency for IconFactory; Added parameter/return type hints

mjansenDatabay committed:
Manual Mail Templates: Default selection for survey

mjansenDatabay committed:
Manual Mail Templates: Default selection for survey

xus committed:
FIlter by available objects type

mjansenDatabay committed:
Terms of Service: Displayed error for self registration issues

mjansenDatabay committed:
CoPage: ilCtrl handling for linked user profiles in kiosk mode scenarios

xus committed:
Filter by status + lang vars

xus committed:
remove dummy line

xus committed:
object type language var

smeyer-ilias committed:
Merge pull request #1170 from studer-raimann/fix/23521/5-3/ilRepUtil_deleteObjectType_Trash

Fix ilRepUtil::deleteObjectType not correctly work with objects in trash

Commit:
Merge pull request #1232 from mjansenDatabay/copage-kiosk-mode-linked-user-profile

CoPage: ilCtrl handling for linked user profiles in kiosk mode scenarios

Commit:
Merge pull request #1207 from leifos-gmbh/precond54

Condition Service: Preconditions by Container

mjansenDatabay committed:
Merge branch 'trunk' into feature-content-page-kiosk-mode

mjansenDatabay committed:
Merge pull request #1227 from leifos-gmbh/5_4_customicon_references

5 4 customicon references

alex40724 committed:
removed custom dbupdate

alex40724 committed:
added missing update step

alex40724 committed:
move new entries to bottom

mjansenDatabay committed:
Merge branch 'trunk' into feature-mail-templates-default

mjansenDatabay committed:
Merged latest trunk changes

mjansenDatabay committed:
Merge pull request #1229 from mjansenDatabay/feature-mail-templates-default

Mail: Default selection for text templates

mjansenDatabay committed:
Custom Icons: PHP 7

alex40724 committed:
merged learning history, step i

alex40724 committed:
merged learning history, step ii

alex40724 committed:
merged learning history, step iii

alex40724 committed:
merged learning history, step iv

alex40724 committed:
Merge branch 'trunk' of https://github.com/ILIAS-eLearning/ILIAS into trunk

smeyer-ilias committed:
Fixed membership gui

smeyer-ilias committed:
LDAP auth

smeyer-ilias committed:
Fixed presentation of custom icons for references

smeyer-ilias committed:
Fixed presentation of custom icons for references

smeyer-ilias committed:
Merge branch '5_4_customicon_references' of https://github.com/leifos-gmbh/ILIAS into 5_4_customicon_references

mjansenDatabay committed:
Terms of Service: Added component based logging

xus committed:
section for edit keywords form

mjansenDatabay committed:
Terms of Service: Fixed translation

alex40724 committed:
fixed #23711: Exc Import

shecken committed:
Custom metadata in settings tab

alex40724 committed:
fixed bug #23609: Whoops\Exception\ErrorException thrown with message A non-numeric value encountered

Amstutz committed:
UI Inputs: Removed typo from comment, added a clarification.

mjansenDatabay committed:
Terms of Service: Used 'Interruptive Modal' for single document deletion process

xus committed:
Hide news before date

Amstutz committed:
UI Modal: Animation fix of #21448

mjansenDatabay committed:
Course/Mail: Fixed #23734

Amstutz committed:
System Styles: Fix of 22564

Commit:
[Learning History] Use composer autoload approach instead including files

Commit:
Renaming variable to avoid confusion

Commit:
Fix wrong declared class hint

bheyser committed:
reintegrated evaluation for all users subtab in statistic tab

xus committed:
Hide by date as a setting for block and timeline.

shecken committed:
IAss: Viewcontrols and filter by mode for members

Commit:
Merge pull request #1234 from legionth/fix-class-hint

[Learning History] Fix wrong declared class hint "iLLanguage" in abstract class

Commit:
Merge pull request #1233 from legionth/learning-history-composer-approach

[Learning History] Use composer autoload approach instead including files

shecken committed:
IAss: Download file from members table content

shecken committed:
IAss: Added some comments

xus committed:
Ass glyph+lang vars+select str.

bheyser committed:
Merge branch 'trunk' of https://github.com/ILIAS-eLearning/ILIAS into trunk

shecken committed:
IAss: Use protected function for txt

shecken committed:
IAss: Set target for sortation control

shecken committed:
IAss: Sort member table data within db query

mjansenDatabay committed:
Object Service: Added 'description' to history when objects are completely deleted from system (e.g. can used for for persisted badges, certificates, etc.)

mjansenDatabay committed:
Object Service: Renamed 'desc' to 'description' in database table 'object_data_del'

rsheer committed:
Added functions to hide object with special chars in title from webdav

Amstutz committed:
System Styles: Fix of #23243

Amstutz committed:
System Styles: Fix of #21699

mjansenDatabay committed:
Content Page: Replaced object type icon

Amstutz committed:
ilTemplate: Templates for Substyles

mjansenDatabay committed:
Merge pull request #1236 from mjansenDatabay/improvement-add-description-to-object-history

Object Service: Added 'description' to history when objects are compl…

Amstutz committed:
System Styles: Fixed lng scope

Amstutz committed:
System Styles: Fixed dep. Check

xus committed:
get parent gui class via ref_id instead of cmdClass

mjansenDatabay committed:
Mail: Renamed command

Amstutz committed:
System Styles: Fix of 23328

mjansenDatabay committed:
Mail: Added missing language variable

mjansenDatabay committed:
Mail: Renamed method, fixed different return types of method, raised and handled exceptions

mjansenDatabay committed:
Mail: Added single download for draft files

alex40724 committed:
fixed autoloader class map

mjansenDatabay committed:
Mail: Renamed bad method name / Added zip download for attachments of mail drafts

mjansenDatabay committed:
Merge pull request #1213 from mjansenDatabay/feature-mail-improvements-54

Mail: Folder table improvements

xus committed:
Merge branch '54_keywords_materials_settings' into trunk

bheyser committed:
usibility fix for streamlining information presentation

bheyser committed:
Merge branch 'trunk' of https://github.com/ILIAS-eLearning/ILIAS into trunk

mjansenDatabay committed:
Object Service/SCORM: Fixed PHP 7 type hint issues

mjansenDatabay committed:
Merge branch 'trunk' of https://github.com/ILIAS-eLearning/ILIAS into trunk

mjansenDatabay committed:
SAML: Logged warning instead of throwing exception

Amstutz committed:
UI Icons: Fixed Intends

mjansenDatabay committed:
SAML: Fixed type

chfsx committed:
FileUpload fix

mjansenDatabay committed:
UI Sortation: Fixed PhpDoc

xus committed:
Tests+mock factory+remove style from tpl

xus committed:
Add todos

xus committed:
remove print_r from test

mjansenDatabay committed:
Merge remote-tracking branch 'mjansen-ilias/trunk' into feature-content-page-kiosk-mode

mjansenDatabay committed:
Merge remote-tracking branch 'mjansen-ilias/trunk' into criteria-terms-of-service

# Conflicts:
#	setup/sql/dbupdate_04.php

mjansenDatabay committed:
Content Page: Kiosk mode dependency handling / Some method extractions

mjansenDatabay committed:
Merge remote-tracking branch 'mjansen-ilias/trunk' into criteria-terms-of-service

# Conflicts:
#	setup/sql/dbupdate_04.php

smeyer-ilias committed:
Added udf plugin slot

alex40724 committed:
feature: Multilingual Page Editor for Container Objects

xus committed:
tests certificate, progressmeter, actions

xus committed:
Tests standard card

xus committed:
remove typo

alex40724 committed:
feature: Assignment Type Wiki

mjansenDatabay committed:
Object Service: Fixed wrong condition in database update step

alex40724 committed:
migrated info box to ks message box

alex40724 committed:
Merge branch 'trunk' of https://github.com/ILIAS-eLearning/ILIAS into trunk

mjansenDatabay committed:
Merge branch 'trunk' into feature-content-page-kiosk-mode

mjansenDatabay committed:
Merge branch 'trunk' into criteria-terms-of-service

mjansenDatabay committed:
Terms of Sercice: Reindexed database update steps

mjansenDatabay committed:
Merge pull request #1212 from mjansenDatabay/criteria-terms-of-service

User: Criteria-based »User Agreement« documents

alex40724 committed:
features: timeline: comments, media, liking, notification on interactions

mjansenDatabay committed:
Content Page: Fixed missing user instance for 'Kiosk Mode'

mjansenDatabay committed:
Content Page: Fixed wrong return value

mjansenDatabay committed:
Content Page: Displayed success message after status change

Commit:
Add template via dependency injection to create unit tests in derived providers

mjansenDatabay committed:
Content Page: Fixed wrong return value

alex40724 committed:
feature: Media Pools: Multiple Media File Import, Filter for inserting media objects from media pool, Custom Metadata in Media Pool

alex40724 committed:
feature: News Items from Plugins (i.e. Repository Objects, ...)

alex40724 committed:
features: Booking Pool: Get additional user information in the booking overview / Reminders and Notifications in Booking Pool

klees committed:
removed duplicate db-update step

Commit:
Merge pull request #1238 from legionth/dependency-injection-template

[Learning History] Add template via dependency injection to create unit tests in derived providers

alex40724 committed:
fixed deck of repo cards view

alex40724 committed:
fixed deck tests and former ->cards() calls (refactored to ->cards()->standard()

alex40724 committed:
fixed image action test

alex40724 committed:
fixed card tests

alex40724 committed:
move colors to variables

mjansenDatabay committed:
Merge branch 'trunk' into feature-content-page-kiosk-mode

mjansenDatabay committed:
Merge pull request #1230 from mjansenDatabay/feature-content-page-kiosk-mode

Content Page: Kiosk Mode

mjansenDatabay committed:
Re-built classmap for autoloading purposes

mjansenDatabay committed:
Forum: Fixed #23760

mjansenDatabay committed:
Forum: Fixed #23760

mjansenDatabay committed:
Forum: Fixed #23757

mjansenDatabay committed:
Merge pull request #1206 from mjansenDatabay/bugfix-abstract-static-methods

Block: Fixed abstract static method declarations (E_STRICT error)

tfamula committed:
Added tests + some fixes

mjansenDatabay committed:
Block: Fixed signature

mjansenDatabay committed:
Block: Fixed magic static call

alex40724 committed:
fixed tests

klees committed:
IAss: 'view record' action not needed anymore

klees committed:
IAss: minor simplification

klees committed:
IAss: some cleanup in presentation table

klees committed:
IAss: fixed lang vars.

klees committed:
SP: consider 'no_type' in new form implementation

klees committed:
SP: fixed update after new form implementation

bheyser committed:
streamlined info/failure messages with nowadays link/button presentations following the guidelines for use of links and buttons witin messages

bheyser committed:
Merge branch '54taMessageBoxes' into trunk

xus committed:
[Fix 23755] Workaround for news context block

klees committed:
UI Outlined Icon: Wording

Co-Authored-By: Amstutz <timon.amstutz@gmail.com>

klees committed:
UI Outlined Icon: Wording

Co-Authored-By: Amstutz <timon.amstutz@gmail.com>

klees committed:
UI Outlined Icon: Wording

Co-Authored-By: Amstutz <timon.amstutz@gmail.com>

Amstutz committed:
UI Outlined Icons: Fixed Intents and Return Type

Amstutz committed:
UI Outlined Icons: Completed Set of outlined Icons.

KissDaniGH committed:
HUN 2018-10-23

alex40724 committed:
switched to standard icons

Amstutz committed:
Merge branch 'trunk' into ui_card_extended

Amstutz committed:
Merge pull request #1194 from leifos-gmbh/ui_card_extended

UI Card extension

mjansenDatabay committed:
Merge pull request #1205 from mjansenDatabay/bugfix-abstract-static-methods-components

Component: Fixed abstract static method declarations (E_STRICT error)

xus committed:
Merge branch 'trunk' into 54_phpspreadsheet

xus committed:
Merge pull request #1162 from leifos-gmbh/54_phpspreadsheet

Migration from PHPExcel(deprecated) to PhpSpreadsheet

mjansenDatabay committed:
HTML Purifier: Updated to latest release

mjansenDatabay committed:
PHPMailer: Updated to latest release

mjansenDatabay committed:
Content Page: Maintainer information

Commit:
Fix setup tools tab not active on some commands

mjansenDatabay committed:
Course/Certificate: Fixed wrong subtab activation

mjansenDatabay committed:
Merge pull request #1243 from mjansenDatabay/bugfix-crs-subtab-activation-cert

Course/Certificate: Fixed wrong subtab activation

mjansenDatabay committed:
Composer Dependencies: Fixed classmap

mjansenDatabay committed:
Mail: Removed require_once directives

smeyer-ilias committed:
Merge pull request #1241 from studer-raimann/fix/0023784/release_5-3/copy-container-with-glossary

0023784: Copying a container with a glossary gets stuck with fatal er…

alex40724 committed:
changed spacing

mjansenDatabay committed:
SAML: Updated to latest release

mjansenDatabay committed:
SAML: Updated to latest release

alex40724 committed:
fixed 23797: Forum: Can't access forum, ilCtrl issues with news side block

alex40724 committed:
fixed #23795: UI Card Extension CSS Broken for Default Icons

alex40724 committed:
removed unused classes

alex40724 committed:
fixed tests

alex40724 committed:
re-added responsive option

klees committed:
Merge pull request #1228 from Amstutz/UI_Outlined_Icon

UI Outlined Icon

chfsx committed:
isNumeric costraint in Numeric Item

rsheer committed:
WebDAV: Implemented delete function for dav objects

klees committed:
Merge pull request #1247 from studer-raimann/fix/0023811/5-4/numeric-input-validation

isNumeric costraint in Numeric Item

Amstutz committed:
Merge pull request #1246 from leifos-gmbh/fix_23795_ui_card_icons

fixed #23795: UI Card Extension CSS Broken for Default Icons

mjansenDatabay committed:
Merge pull request #1244 from mjansenDatabay/simplesaml-dependency-update

 SAML: Updated to latest release of SimpleSAMLphp

alex40724 committed:
fixed test; added type hints

alex40724 committed:
features: usability portfio content tab; creating portfolio from template, list of portfolios, skill overview

alex40724 committed:
features: Set Portfolio Online Message in Share-tab, Prompting to Personal Profile in First Access, Extension of Learning History: Encouraging statement and Starting Point

alex40724 committed:
features: Portfolio: Declaration of Authorship, Sum up different perspectives to learning in a central area; Streamlining Message Boxes

alex40724 committed:
features: Page Element for Portfolios: Learning History

Amstutz committed:
Merge branch 'trunk' into 5_4_ks_toggle_button

smeyer-ilias committed:
Merge remote-tracking branch 'leifos/5_4_customicon_references' into trunk

Amstutz committed:
Merge pull request #1098 from leifos-gmbh/5_4_ks_toggle_button

KS: Toggle Button

smeyer-ilias committed:
Merge branch 'trunk' of https://github.com/ILIAS-eLearning/ILIAS into trunk

Amstutz committed:
UI Tests: Fixed Toggle Tests

mjansenDatabay committed:
Chat: Updated documentation in regards of dependency management

mjansenDatabay committed:
Chat: Updated documentation in regards of dependency management

klees committed:
Merge pull request #1242 from studer-raimann/fix/0023798/5-3/fix_setup_tools_tab_not_active_on_some_commands

Fix setup tools tab not active on some commands

mjansenDatabay committed:
Merge pull request #1250 from mjansenDatabay/chat-dependencies-docu-yarn

Chat: Updated documentation in regards of dependency management

xus committed:
Fix Exercise List All submissions UI Card

klees committed:
IAss: some formatting in participants table

klees committed:
Merge pull request #1237 from shecken/trunk_pt_iass

Presentation Table for Individual Assessments

xus committed:
merge exc random teams
